### PR TITLE
Turn `cvc5_proof_rule.cpp` into a template.

### PR DESCRIPTION
**Pull request by abdoo8080**
"The only file that should be (manually) updated in the source is the public header file `cvc5_proof_rule.h`."

### CODE REVIEW

File: src/rewriter/CMakeLists.txt [hunk 1]
```diff
@@ -50,13 +50,19 @@ libcvc5_add_sources(GENERATED
   ${REWRITE_OUTPUT_FILES}
 )
 
-# If the generated files differ from the source files, they should be copied to
+# If the generated header differs from the source header, it should be copied to
```
Comment (ajreynol) [inline src/rewriter/CMakeLists.txt:53]: Should `cvc5_proof_rule.cpp` be removed from the block below? (line 56)

Comment (abdoo8080) [reply src/rewriter/CMakeLists.txt]: No. `cvc5_proof_rule.cpp` was added to `LIBCVC5_GEN_SRCS` instead. I moved the comment there to explain why the generated `cvc5_proof_rule.h` is not tracked.

Comment (abdoo8080) [reply src/rewriter/CMakeLists.txt]: No. `cvc5_proof_rule.cpp` was added to `LIBCVC5_GEN_SRCS` instead. I moved the comment there to explain why the generated `cvc5_proof_rule.h` is not tracked.

**Review by ajreynol — APPROVED** [review]
LGTM

**Pull request merged**